app/main.py

```diff
@@ -1,3 +1,113 @@
+from __future__ import annotations
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union[Distance, float, int]) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        elif isinstance(other, (int, float)):
+            return Distance(self.km + float(other))
+        else:
+            raise TypeError(
+                f"Unsupported operand type(s) for +: 'Distance' and "
+                f"'{type(other).__name__}'"
+            )
+
```

Comment:
Code duplication: You have similar code blocks in the `__add__`, `__iadd__`, `__lt__`, `__gt__`, `__eq__`, `__le__`, `__ge__` methods. You could reduce this duplication by creating a helper method that checks the type of `other` and converts it to the appropriate type if necessary. This would make your code cleaner and easier to maintain.